src/app/widgets/popup/popup-3fgl.ts

```diff
@@ -68,14 +51,25 @@ export class Popup3FGL {
           </table>
 
         </div>
-
+        ` + this.getSourceDetailUrl() + `
     `;
   }
 
   round(val) {
     return (Math.round(val * 100) / 100).toFixed(2);
   }
 
+  getSourceDetailUrl() {
```

Comment:
Maybe have a separate function that generates just the URL string.
Not real code, just roughly, the return value would be a string with the relative URL:
```
#cat/{source.cat_name}/{source.source_id}`
```
The rest of these functions, the HTML with `<div class='popup-link'>` is duplicated for each cat. This 4x code duplication could be easily avoided, no?